src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -376,43 +376,102 @@ internal static IEnumerable<CommandParameterInternal> Splat(object splattedValue
                         fromSplatting: true);
                 }
             }
+            else if (splattedValue is IEnumerable enumerableValue)
+            {
+                foreach (CommandParameterInternal param in SplatIEnumerable(enumerableValue, splatAst, markUntrustedData))
+                {
+                    yield return param;
+                }
+            }
             else
             {
-                IEnumerable enumerableValue = splattedValue as IEnumerable;
-                if (enumerableValue != null)
+                yield return CommandParameterInternal.CreateArgument(splattedValue, splatAst);
+            }
+        }
+
+#nullable enable
+        private static IEnumerable<CommandParameterInternal> SplatIEnumerable(
+            IEnumerable enumerable,
+            Ast splatAst,
+            bool markUntrustedData)
+        {
+            var enumerator = enumerable.GetEnumerator();
+            try
+            {
+                while (enumerator.MoveNext())
                 {
-                    foreach (object obj in enumerableValue)
+                    object obj = enumerator.Current;
+                    if (markUntrustedData)
                     {
-                        if (markUntrustedData)
+                        ExecutionContext.MarkObjectAsUntrusted(obj);
+                    }
+
+                    if (TryGetSplattedCommandParameterInfo(obj, out string? paramText, out string? paramName))
+                    {
+                        // Parameters that end with : and have a subsequent value should be treated as parameter
+                        // and argument. This allows passing through a splatted switch with an explicit value.
+                        // Without this the switch and switch value will be treated as separate arguments.
+                        if (paramText.EndsWith(':') && enumerator.MoveNext())
                         {
-                            ExecutionContext.MarkObjectAsUntrusted(obj);
+                            yield return CommandParameterInternal.CreateParameterWithArgument(
+                                parameterAst: splatAst,
+                                parameterName: paramName,
+                                parameterText: paramText,
+                                argumentAst: splatAst,
+                                value: enumerator.Current,
+                                spaceAfterParameter: false,
+                                fromSplatting: true);
+                        }
+                        else
+                        {
+                            yield return CommandParameterInternal.CreateParameter(paramName, paramText, splatAst);
                         }
-
-                        yield return SplatEnumerableElement(obj, splatAst);
                     }
-                }
-                else
-                {
-                    yield return SplatEnumerableElement(splattedValue, splatAst);
```

Comment:
This non IDictionary non IEnumerable case originally did the same check for the note property but I've found it would never have applied as it also checked if the same value was a string which would have been treated as an IEnumerable above anyway. So in all cases this final one is just a single value to provide as a single argument.